--- procps-3.2.7/proc/sysinfo.c	2006-06-25 08:41:48.000000000 +0200
+++ src/proc/sysinfo.c	2009-03-27 00:34:45.000000000 +0100
@@ -882,15 +882,7 @@
   close(fd);
   if(rc<3) goto out;
   pidbuf[rc] = '\0';
-  rc = strtol(pidbuf,&endp,10);
-  if(rc<42) goto out;
-  if(*endp && *endp!='\n') goto out;
-  rc--;  // the pid_max value is really the max PID plus 1
-  ret = 0;
-  while(rc){
-    rc /= 10;
-    ret++;
-  }
+  ret = strlen(pidbuf);
 out:
   return ret;
 }
