--- libnl-1.1/lib/route/addr.c	2008-01-14 16:48:45.000000000 +0100
+++ src/lib/route/addr.c	2010-07-01 00:12:55.586701922 +0200
@@ -1090,8 +1090,7 @@
 	.oo_compare		= addr_compare,
 	.oo_attrs2str		= addr_attrs2str,
 	.oo_id_attrs		= (ADDR_ATTR_FAMILY | ADDR_ATTR_IFINDEX |
-				   ADDR_ATTR_LOCAL | ADDR_ATTR_PREFIXLEN |
-				   ADDR_ATTR_PEER),
+				   ADDR_ATTR_LOCAL | ADDR_ATTR_PREFIXLEN),
 };
 
 static struct nl_af_group addr_groups[] = {
