diff -ru gst-plugins-bad-0.10.18.orig/gst-libs/gst/video/gstbasevideocodec.h gst-plugins-bad-0.10.18/gst-libs/gst/video/gstbasevideocodec.h
--- gst-plugins-bad-0.10.18.orig/gst-libs/gst/video/gstbasevideocodec.h	2010-05-16 02:49:52.515489286 +0200
+++ gst-plugins-bad-0.10.18/gst-libs/gst/video/gstbasevideocodec.h	2010-05-16 02:50:41.345471555 +0200
@@ -26,7 +26,7 @@
 #endif
 
 #include <gst/gst.h>
-#include <gst/video/gstbasevideoutils.h>
+#include "gstbasevideoutils.h"
 
 G_BEGIN_DECLS
 
diff -ru gst-plugins-bad-0.10.18.orig/gst-libs/gst/video/gstbasevideodecoder.h gst-plugins-bad-0.10.18/gst-libs/gst/video/gstbasevideodecoder.h
--- gst-plugins-bad-0.10.18.orig/gst-libs/gst/video/gstbasevideodecoder.h	2010-05-16 02:49:52.515489286 +0200
+++ gst-plugins-bad-0.10.18/gst-libs/gst/video/gstbasevideodecoder.h	2010-05-16 02:50:49.015476617 +0200
@@ -25,7 +25,7 @@
 #warning "You can define GST_USE_UNSTABLE_API to avoid this warning."
 #endif
 
-#include <gst/video/gstbasevideocodec.h>
+#include "gstbasevideocodec.h"
 
 G_BEGIN_DECLS
 
diff -ru gst-plugins-bad-0.10.18.orig/gst-libs/gst/video/gstbasevideoencoder.h gst-plugins-bad-0.10.18/gst-libs/gst/video/gstbasevideoencoder.h
--- gst-plugins-bad-0.10.18.orig/gst-libs/gst/video/gstbasevideoencoder.h	2010-05-16 02:49:52.515489286 +0200
+++ gst-plugins-bad-0.10.18/gst-libs/gst/video/gstbasevideoencoder.h	2010-05-16 02:50:58.725489768 +0200
@@ -25,8 +25,8 @@
 #warning "You can define GST_USE_UNSTABLE_API to avoid this warning."
 #endif
 
-#include <gst/video/gstbasevideocodec.h>
-#include <gst/video/gstbasevideoutils.h>
+#include "gstbasevideocodec.h"
+#include "gstbasevideoutils.h"
 
 G_BEGIN_DECLS
 
diff -ru gst-plugins-bad-0.10.18.orig/gst-libs/gst/video/gstbasevideoparse.h gst-plugins-bad-0.10.18/gst-libs/gst/video/gstbasevideoparse.h
--- gst-plugins-bad-0.10.18.orig/gst-libs/gst/video/gstbasevideoparse.h	2010-05-16 02:49:52.515489286 +0200
+++ gst-plugins-bad-0.10.18/gst-libs/gst/video/gstbasevideoparse.h	2010-05-16 02:51:24.456653077 +0200
@@ -25,8 +25,8 @@
 #warning "You can define GST_USE_UNSTABLE_API to avoid this warning."
 #endif
 
-#include <gst/video/gstbasevideocodec.h>
-#include <gst/video/gstbasevideoutils.h>
+#include "gstbasevideocodec.h"
+#include "gstbasevideoutils.h"
 
 G_BEGIN_DECLS
 
